### PR TITLE
hedge_ratio.py

### PR DESCRIPTION
**Pull request by Ksteeds**
Commencing code.

### CODE REVIEW

**Comment by Ksteeds** [comment]
Work-in-progress.

File: macrosynergy/panel/hedge_ratio.py [hunk 1]
```diff
@@ -1,39 +1,274 @@
...
+        day, and the gap between each date is delimited by the frequency parameter.
+    """
 
-def hedge_ratio(df: pd.DataFrame, xcats: List[str] = None, cids: List[str] = None,
-                hedge_return: str = None,
-                start: str = None, end: str = None,
-                meth: str = 'ols', oos: bool = True, rfreq: str = 'm',
-                minobs: int = 24,
-                blacklist: dict = None):
+    dates = pd.date_range(start_date, end_date, freq=refreq)
```
Comment (rsueppel) [inline macrosynergy/panel/hedge_ratio.py:25]: The ratio needs to be calculated based on daily data (for now). The rebalancing frequency only affects the intervals of recalculation.

File: macrosynergy/panel/hedge_ratio.py [hunk 2]
```diff
@@ -1,39 +1,274 @@
...
+    break up the respective return series, which are defined daily, into the re-estimated
+    frequency paradigm. To achieve this ensure the dates produced fall on business days,
+    and will subsequently be present in the return-series dataframes.
+
+    :param <pd.Timestamp> start_date:
+    :param <pd.Timestamp> end_date:
+    :param <str> refreq:
+
+    return <List[pd.Timestamp]>: List of timestamps where each date is a valid business
+        day, and the gap between each date is delimited by the frequency parameter.
```
Comment (rsueppel) [inline macrosynergy/panel/hedge_ratio.py:22]: Does this mean this returns the dates of the rebalancing days?

**Comment by Ksteeds** [comment]
Solving ToDos.

**Comment by Ksteeds** [comment]
Believe code is ready for testing.

**Comment by Ksteeds** [comment]
Updates following code review.

**Comment by Ksteeds** [comment]
Believe latest ToDos have been addressed - minor changes.

**Review by rsueppel — APPROVED** [review]

**Pull request merged**